src/app.routes.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import permission from './path.json';
+import { Redirect, Route } from "react-router-dom"; 
+
+const ProtectedRoute = ({ component: Component, role, path, ...rest}) => {
+    const validatePath = (path) => {
+        const index = permission.role.indexOf(path);
+        return (index !== -1)
+    }
+    const isAllowed = validatePath();
```

Comment:
Do you have to call it here and assign it to variable and use it in return or call validatepath function in return directly